### ISSUE TITLE
Generate signed URL for Google Cloud Storage with additional parameters in the URL

### ISSUE THREAD

**Issue by ihsanhaikalz** [googleapis/java-storage#136]
Is it possible to generate a signed URL for Google Cloud Storage with additional parameters made by us (such as X-Goog-OurParameter=Hello)? If so, will it be shown in the Storage log or Stackdriver Log?

**Comment by frankyn** [comment]
@houglum is logging Signed URL requests part of scope for Audit Logs?

**Comment by frankyn** [comment]
Friendly ping @houglum

**Comment by houglum** [comment]
A signed request is like a "regular" (i.e. OAuth2-token-authenticated) request, except that it's authenticated as the user that the signature's key belongs to -- it's as if **that user** made the request directly -- this applies to the logging logic too.  So if you couldn't do something (e.g. supply a random query param) with a "regular" request, then you couldn't do it with a signed request either.

**Comment by frankyn** [comment]
Thanks @houglum, @ihsanhaikalz did find that information is logged in Access logs but not Audit Logs, is that also WAI?

@ihsanhaikalz you should be able to set custom query parameters now.
https://googleapis.dev/java/google-cloud-storage/latest/com/google/cloud/storage/Storage.SignUrlOption.html#withQueryParams-java.util.Map-

**Comment by houglum** [comment]
When you say "that information", do you mean the entire request is logged in an access log entry, but not an audit log entry?  Or that certain attributes of the request are visible in the access log entry but not the audit log entry?

If you meant the former, we don't publish audit logs for a request if the access was granted via the object's ACLs, rather than the parent bucket's IAM policy.

**Comment by ihsanhaikalz** [comment]
@frankyn Thanks very much for the info! I will try it out later and let you know
@houglum Actually  the signed URL or parts of the URL, such as https://storage.googleapis.com/example-bucket/cat.jpeg?X-Goog-Algorithm=
GOOG4-RSA-SHA256&......, is not logged in the Stackdriver log but it is logged in Access Storage log. As a customer, I think it will be interesting to know how does the user access the object in the bucket.

**Comment by frankyn** [comment]
I think @houglum is following up on this, but not sure if there's an update from his side.

**Comment by houglum** [comment]
Discussed this a bit today; going to let Frank try to make a clean repro of this with screenshots, provided-vs-expected values, and the whole shebang. This will give us a bug to track internally to remedy any issues.

**Comment by frankyn** [comment]
Hi @houglum,

I repro'd the issue by doing the following (You need to use an external GCP Project as internal GCP projects will not generate Access Logs):

1. [Enabling Cloud Audit](https://cloud.google.com/logging/docs/audit/configure-data-access#config-console).
2. [Enabling Access logs](https://cloud.google.com/storage/docs/access-logs). 
3. I exercised the [Python sample](https://cloud.google.com/storage/docs/access-control/signing-urls-with-helpers#storage-signed-url-object-python) to generate a GET Signed URL.

```python
from google.cloud import storage
import datetime


def generate_download_signed_url_v4(bucket_name, blob_name):
    """Generates a v4 signed URL for downloading a blob.

    Note that this method requires a service account key file. You can not use
    this if you are using Application Default Credentials from Google Compute
    Engine or from the Google Cloud SDK.
    """
    # bucket_name = 'your-bucket-name'
    # blob_name = 'your-object-name'

    storage_client = storage.Client()
    bucket = storage_client.bucket(bucket_name)
    blob = bucket.blob(blob_name)

    url = blob.generate_signed_url(
        version="v4",
        # This URL is valid for 15 minutes
        expiration=datetime.timedelta(minutes=15),
        # Allow GET requests using this URL.
        method="GET",
    )

    print("Generated GET signed URL:")
    print(url)
    print("You can use this URL with any user agent, for example:")
    print("curl '{}'".format(url))
    return url
```
4. I executed the request ~10x (no longer works):
```bash
curl "https://coderfrank-website.storage.googleapis.com/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=my-access-logs-project%40appspot.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T024857Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=45fa66942d197b8f3b6e17cd9f44347796a6a962bcb9473259784abca91b21b9b9144e5dd4fc9aab75e35759d504f0ea3a40d725e950f8f78a40de3162fe9554ab4bafd53b92cb55680c124def64954fcfb18e8eb243404f4f2f76a21a0209833202d458f6a3f11f259e9dddb62b4d2b96ba3ade3b2c521702abd6751d85f879e6ba799b88d27b9f8f0c1ef7cf26bd6a18f9d1b517942dbd5455a0be7c8c59e7a8c975ff8df6c3b20edd1b27aab1787dbc17dff12737ef8200e905bf6dcecf6c784142558a0ebca7a703aa4cfe16df14f8fb88469f6691c346deb45594f8dbc1318b1195835b01e31d6f409f08d053f5aa91f8ffdec3afe5ded9ffa15e68052c"
```

5. After waiting for data to be generated in my Access Logs bucket (takes about ~1.5 hours). In my logs bucket I found the following object `logs__usage_2020_05_06_02_00_00_077f2cf1be89d037ba_v0` with contents:

```
"time_micros","c_ip","c_ip_type","c_ip_region","cs_method","cs_uri","sc_status","cs_bytes","sc_bytes","time_taken_micros","cs_host","cs_referer","cs_user_agent","s_request_id","cs_operation","cs_bucket","cs_object"
"1588731313422244","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89985","200","0","761","35000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUn5wIg5PWltdy4N70lQTS_LuC6gjrlhwZKDw5ennYgnMfRcIVMooH1bCa-IHUJD1ApfyD_gw-OePBnGnU_8LGE","GET_Object","coderfrank-website","index.html"
"1588731311966915","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89985","200","0","761","35000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUnRDPftqMF3xJZojaDgZLpX4b6IDKEiPOXyeEA6IBK1VZQ72NuJ4PcMe-CSguEfWCrTbHqOr7WgdNPvPaCNgOg","GET_Object","coderfrank-website","index.html"
"1588731314158422","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89985","200","0","761","34000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUnNR0A2Ulx56Yy4Yb_HkhgpGfOQlYTVhR2_TnzNfxrPEgU4BDrCoeRtzEvWJE39Tp47eARH1IXOhxjPOssXQ3w","GET_Object","coderfrank-website","index.html"
"1588731340756329","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89986","403","0","801","261000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUkU3cALtDwIkQ8792LkeehTIh8HdCRbCfdaJA8aKflaixEt96tZt6P-Ob6AhIjIUKBvHSEnVU_vgv9jVQZ1phM","GET_Object","coderfrank-website","index.html"
"1588733273635349","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89986","403","0","801","501000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUmfeyvwBtOdwkkEjwf3BV2g0vEIH1wwIDPZAHKAHT1x07dZpFdwOQ_nrE41-EF1VA-zF5ZoeG8xTKIcauEHjWs","GET_Object","coderfrank-website","index.html"
"1588733283905037","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89985","200","0","761","32000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUlyMNPopNCLpA9RJ742N6pRAanBv3S-uoZctapO69neTDobl9lcHHshgkFUmaJRdSVX9TEqDMPTpsiydxAxgRQ","GET_Object","coderfrank-website","index.html"
"1588733283295366","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89985","200","0","761","36000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUlBUFlds7pHHfXuDlw-aN3j-GkxChkp9encHaAGW5wbPnV1WDMXaY8U-LdIIzhHiwSZOvDI9CJmaUJjkSxhxgU","GET_Object","coderfrank-website","index.html"
"1588733284536158","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=storage-python%40spec-test-ruby-samples.iam.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T020710Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=4dcf5ead000de13d0ed2384a2e0c2b5ad2db4f6550f348d109b0f30163d1548f3ddb56b35c3a1afd0cd63514654975ed4c3c6b180c257bd4d8eb855e3316258ad0784b1c2a644b31877eb787a265e661942b28de2321f15a470f68588b9a98acf03abd76fd976974cd6ab22ae38124c75982d489790b831ed730cf3351fa0336ad6f76f310ceebb3dd05e26e9ea86ceb8d8241fb91a0c95b6c8bdd91f852bd6c24bcdd4d6d8e3f15a09e9c60d9ebd6dfd8915a26a05832e943f0cfbb07f50f933c199e08adc0789b75a289503a3a9caa0019ab388bd8dbd79f9ac05dab621d3eecaf56719eba883d531f1ea01702e94b980b42a72581c1de69b41c4e76d89985","200","0","761","38000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUlU8U4cvLCMw4Sfxk6GQsJPWqcqVCyqXh7dgVA8Q2ADLq0wqK47i1WTBNfI_zgRYSPfLuisL_sXOPflGKuO1ps","GET_Object","coderfrank-website","index.html"
"1588733353589420","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=my-access-logs-project%40appspot.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T024857Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=45fa66942d197b8f3b6e17cd9f44347796a6a962bcb9473259784abca91b21b9b9144e5dd4fc9aab75e35759d504f0ea3a40d725e950f8f78a40de3162fe9554ab4bafd53b92cb55680c124def64954fcfb18e8eb243404f4f2f76a21a0209833202d458f6a3f11f259e9dddb62b4d2b96ba3ade3b2c521702abd6751d85f879e6ba799b88d27b9f8f0c1ef7cf26bd6a18f9d1b517942dbd5455a0be7c8c59e7a8c975ff8df6c3b20edd1b27aab1787dbc17dff12737ef8200e905bf6dcecf6c784142558a0ebca7a703aa4cfe16df14f8fb88469f6691c346deb45594f8dbc1318b1195835b01e31d6f409f08d053f5aa91f8ffdec3afe5ded9ffa15e68052c","200","0","761","31000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUlI2vZXy_tqs6vPLi0Xp5Yshv0nD0V7MMYqrkNLD1JxXHfl2gh0PY3XSX65ovOzaHRbWqEjx45pTZU835mvuho","GET_Object","coderfrank-website","index.html"
"1588733353215866","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=my-access-logs-project%40appspot.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T024857Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=45fa66942d197b8f3b6e17cd9f44347796a6a962bcb9473259784abca91b21b9b9144e5dd4fc9aab75e35759d504f0ea3a40d725e950f8f78a40de3162fe9554ab4bafd53b92cb55680c124def64954fcfb18e8eb243404f4f2f76a21a0209833202d458f6a3f11f259e9dddb62b4d2b96ba3ade3b2c521702abd6751d85f879e6ba799b88d27b9f8f0c1ef7cf26bd6a18f9d1b517942dbd5455a0be7c8c59e7a8c975ff8df6c3b20edd1b27aab1787dbc17dff12737ef8200e905bf6dcecf6c784142558a0ebca7a703aa4cfe16df14f8fb88469f6691c346deb45594f8dbc1318b1195835b01e31d6f409f08d053f5aa91f8ffdec3afe5ded9ffa15e68052c","200","0","761","35000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUlir21o5Inv99OglXRntJppFt-3MzTixTM6hjmpJcSB_oVPe2ZZiKFmOW88bWoOam0XrCohUb_dZu1GpRnXiek","GET_Object","coderfrank-website","index.html"
"1588733357040002","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=my-access-logs-project%40appspot.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T024857Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=45fa66942d197b8f3b6e17cd9f44347796a6a962bcb9473259784abca91b21b9b9144e5dd4fc9aab75e35759d504f0ea3a40d725e950f8f78a40de3162fe9554ab4bafd53b92cb55680c124def64954fcfb18e8eb243404f4f2f76a21a0209833202d458f6a3f11f259e9dddb62b4d2b96ba3ade3b2c521702abd6751d85f879e6ba799b88d27b9f8f0c1ef7cf26bd6a18f9d1b517942dbd5455a0be7c8c59e7a8c975ff8df6c3b20edd1b27aab1787dbc17dff12737ef8200e905bf6dcecf6c784142558a0ebca7a703aa4cfe16df14f8fb88469f6691c346deb45594f8dbc1318b1195835b01e31d6f409f08d053f5aa91f8ffdec3afe5ded9ffa15e68052c","200","0","762","35000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUk4omu8a5f7LQhtGRHYDvw48hmSA3g7m0DBvcvnOWmrx_6KlGtIPQKQnEqXMWe-ySHtUKfAIjCy_y9kY4pYHOo","GET_Object","coderfrank-website","index.html"
"1588733357450361","2601:646:9680:6300::12eb","2","","GET","/index.html?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=my-access-logs-project%40appspot.gserviceaccount.com%2F20200506%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20200506T024857Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&extra-parameter=extra-value&x-goog-signature=45fa66942d197b8f3b6e17cd9f44347796a6a962bcb9473259784abca91b21b9b9144e5dd4fc9aab75e35759d504f0ea3a40d725e950f8f78a40de3162fe9554ab4bafd53b92cb55680c124def64954fcfb18e8eb243404f4f2f76a21a0209833202d458f6a3f11f259e9dddb62b4d2b96ba3ade3b2c521702abd6751d85f879e6ba799b88d27b9f8f0c1ef7cf26bd6a18f9d1b517942dbd5455a0be7c8c59e7a8c975ff8df6c3b20edd1b27aab1787dbc17dff12737ef8200e905bf6dcecf6c784142558a0ebca7a703aa4cfe16df14f8fb88469f6691c346deb45594f8dbc1318b1195835b01e31d6f409f08d053f5aa91f8ffdec3afe5ded9ffa15e68052c","200","0","762","33000","coderfrank-website.storage.googleapis.com","","curl/7.67.0,gzip(gfe)","AAANsUnOcuRL2g8QxiRZCAdrbdmPpJmTGxkwFBiJQfXMbfb50atnfqngSXRM0nzf78dK5a-SRLt-6VIGc1FZNz7n-4M","GET_Object","coderfrank-website","index.html"
```

6. Comparing against Cloud Audit Logs:
[gcs_bucket__logs__2020-05-05T09-04.txt](https://github.com/googleapis/java-storage/files/4584795/gcs_bucket__logs__2020-05-05T09-04.txt)


### Conclusion
Cloud Audit Logs doesn't log Signed URL requests, but Access Logs does log the full URI. If the request happens right before Access Logs are generated or expiration is in terms of multiple hours then Signed URLs can be valid.

**Comment by frankyn** [comment]
Hi @ihsanhaikalz,

Chatted with backend team and the consensus is that Audit Logs will not log this level of information for Signed URLs given potential security risks. Access Logs is the only path right now.

Closing issue given this information. Please feel free to reopen if this did not answer your question. Thank you for your patience.

**Comment by houglum** [comment]
Just a heads up, we ended up discovering that the request not showing up in Cloud Audit Logging was due to the object being public (readable/updatable by the "AllUsers" principal).  This is [mentioned in the docs](https://cloud.google.com/storage/docs/audit-logs):

> Cloud Audit Logs does not track access to public objects.

I just verified that accessing an non-public object through a V4 signed URL **does** indeed generate a Cloud Audit Logging data access log entry.  Note, however, that these records **do not show query parameters by default**; you'd have to have [Detailed Audit Logging Mode](https://cloud.google.com/storage/docs/org-policy-constraints#audit-logging) enabled to have those show up.
